core/arch/arm/mm/tee_pager.c

```diff
@@ -260,13 +260,18 @@ static void *pager_add_alias_page(paddr_t pa)
 	struct core_mmu_table_info *ti = &pager_alias_tbl_info;
 	uint32_t attr = TEE_MATTR_VALID_BLOCK | TEE_MATTR_GLOBAL |
 			(TEE_MATTR_CACHE_CACHED << TEE_MATTR_CACHE_SHIFT) |
-			TEE_MATTR_SECURE | TEE_MATTR_PRW;
+			TEE_MATTR_SECURE | TEE_MATTR_PR;
 
 	DMSG("0x%" PRIxPA, pa);
 
 	if (!pager_alias_next_free || !ti->num_entries)
 		panic("invalid alias entry");
 
+#ifndef CFG_CORE_RWDATA_NOEXEC
```

Comment:
I'd rather try to protect as much as possible in the pager. So if no one objects please remove all these ifdefs in this patch.